### ISSUE TITLE
Method missing cop and BasicObject

### ISSUE THREAD

**Issue by fxn** [rubocop-hq/rubocop#4761]
## Expected behavior

Given

```ruby
class C < BasicObject
  def method_missing(*)
    super
  end
end
```

the method missing cop should issue no warning.

## Actual behavior

The method missing cop issues:

    When using method_missing, define respond_to_missing?.

## Steps to reproduce the problem

The example above is self-explanatory.

## Rationale

`BasicObject` instances do not respond to `respond_to?`, nor they respond to `method`. There is no use case for `respond_to_missing?`.

## RuboCop version

```
$ rubocop -V
0.45.0 (using Parser 2.3.3.1, running on ruby 2.2.5 x86_64-linux)
```

**Comment by Drenmi** [comment]
Hm. Calling super should raise an exception, since `BasicObject` has no parent. So that part of the recommendation also doesn't apply. 🤔 Makes sense for this cop to ignore `BasicObject`.

**Comment by fxn** [comment]
But `C` has a parent, and `BasicObject` implements `method_missing`... maybe that one is still good?

**Comment by Drenmi** [comment]
Then I'm not quite following. You mean this:

```ruby
class C < BasicObject
  def method_missing(*)
    super
  end
end
```

is the entire implementation, and `C` is meant to be another `BasicObject`?

**Comment by fxn** [comment]
I mean, since `BasicObject#method_missing` exists, then `super` makes sense because `BasicObject` is the superclass of `C`. In particular delegate error message and exception object.

**Comment by fxn** [comment]
Related https://github.com/ruby/ruby/pull/1701.

**Comment by Drenmi** [comment]
Right. I see the issue now. Will wait for resolution of ruby/ruby#1701.
